src/sage/quadratic_forms/ternary.pyx

```diff
@@ -672,54 +641,45 @@ def _find_all_ternary_qf_by_level_disc(long long N, long long d):
     cdef double a_max
     cdef double r_max
     cdef double b_max
-    cdef long long stu2
-    cdef long long mg
-
 
+    l = []
 
-    l=[]
-
-    if (4*d)%N!=0:
+    if (4*d) % N:
         raise ValueError("There are no ternary forms of this level and discriminant")
     else:
-        m=4*d//N
+        m = 4*d//N
 
-    if (N**2)%d!=0:
+    if (N**2) % d:
         raise ValueError("There are no ternary forms of this level and discriminant")
-    else:
-        mu=N*N//d
 
-    m_2=m%2
-    mu_2=mu%2
+    a = 1
+    a_max = (d/2.)**(1/3.)
+    while a <= a_max:
 
-    a=1
-    a_max=(d/2.)**(1/3.)
-    while a<=a_max:
+        [g, u, v] = xgcd(4*a, m)
+        g1 = (ZZ(g).squarefree_part()*g).sqrtrem()[0]
+        t = 0
+        while t <= a:
 
-        [g,u,v]=xgcd(4*a,m)
-        g1=(ZZ(g).squarefree_part()*g).sqrtrem()[0]
-        t=0
-        while t<=a:
+            alpha = max(a, m/4/a)
+            b = alpha+((((u*t*t//g)-alpha)) % (m//g))
+            b_max = (d/2.0/a)**(1/2.)
+            while b <= b_max:
 
-            alpha=max(a,m/4/a)
-            b=alpha+((((u*t*t//g)-alpha))%(m//g))
-            b_max=(d/2.0/a)**(1/2.)
-            while b<=b_max:
+                s = 0
+                beta = g//gcd(g, 2*t)
+                while s <= a:
 
-                s=0
-                beta=g//gcd(g,2*t)
-                while s<=a:
-
-                    r = -b+((((2*s*t*u//g)+b))%(m//g))
-                    if s*t==0:
-                        r_max=0
+                    r = -b+((((2*s*t*u//g)+b)) % (m//g))
+                    if s*t == 0:
+                        r_max = 0
                     else:
-                        r_max=b
-                    while r<=r_max:
+                        r_max = b
+                    while r <= r_max:
 
-                        if (d-r*s*t+a*r*r+b*s*s)%(4*a*b-t**2)==0:
+                        if (d-r*s*t+a*r*r+b*s*s) % (4*a*b-t**2) == 0:
```

Comment:
spaces around `+` and `- ` ?

Reply:
Je comprends. Je trouvais étrange que tu ne le fasses pas en même temps sur les lignes que tu as modifié. On va donc laisser comme ça.